src/Admin/Product/Attributes/VariationsAttributes.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Class VariationsAttributes
+ *
+ * @package Automattic\WooCommerce\Pinterest\Admin\Product\Attributes
+ */
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Pinterest\Admin\Product\Attributes;
+
+use Automattic\WooCommerce\Pinterest\Admin\Admin;
+use Automattic\WooCommerce\Pinterest\Admin\Input\Form;
+use Automattic\WooCommerce\Pinterest\Product\Attributes\AttributeManager;
+use WC_Product_Variation;
+use WP_Post;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class VariationsAttributes
+ */
+class VariationsAttributes {
+
+	/**
+	 * Admin class.
+	 *
+	 * @var Admin
+	 */
+	protected $admin;
+
+	/**
+	 * Attibute Manager.
+	 *
+	 * @var AttributeManager
+	 */
+	protected $attribute_manager;
+
+	/**
+	 * VariationsAttributes constructor.
+	 *
+	 * @param Admin $admin Admin class.
+	 */
+	public function __construct( Admin $admin ) {
+		$this->admin             = $admin;
+		$this->attribute_manager = AttributeManager::instance();
+	}
+	/**
+	 * Register a service.
+	 */
+	public function register(): void {
+		add_action(
+			'woocommerce_product_after_variable_attributes',
+			function ( int $variation_index, array $variation_data, WP_Post $variation ) {
+				$this->render_attributes_form( $variation_index, $variation );
+			},
+			90,
+			3
+		);
+		add_action(
+			'woocommerce_save_product_variation',
+			function ( int $variation_id, int $variation_index ) {
+				$this->handle_save_variation( $variation_id, $variation_index );
+			},
+			10,
+			2
+		);
+	}
+
+	/**
+	 * Render the attributes form for variations.
+	 *
+	 * @param int     $variation_index Position in the loop.
+	 * @param WP_Post $variation       Post data.
+	 */
+	private function render_attributes_form( int $variation_index, WP_Post $variation ) {
+		/**
```

Comment:
Do we really need to comment on this?

Reply:
It's to clarify the difference between $variation which is an instance of WP_Post and $product which is an instance of WC_Product_Variation. It's not strictly necessary but I copied this code over directly from GLA so I left it as is.